### DISCUSSION TITLE
Error: Could NOT find GLEW

### DISCUSSION THREAD

**Discussion by Freeplayg** [Almamu/linux-wallpaperengine#120] [Q&A]
Unless I'm being dumb because new to linux, I think glew is already installed. If not, can someone please tell how to install it? 
![Screenshot from 2022-06-22 20-42-39](https://user-images.githubusercontent.com/40742947/175183775-7176ca64-9305-40b0-a8fe-690ae9cf6f81.png)

**Comment by OposDev** [comment]
If you're on Debian (Or another OS that utilizes apt) try installing the `libglew-dev` package, this should fix this error :D

**Comment by Freeplayg** [comment]
I'm on Fedora, sadly D:

**Comment by OposDev** [comment]
Maybe Fedora has a similar package in their repositories? Try to find one specified as a lib / dev package.

https://fedora.pkgs.org/36/fedora-aarch64/libGLEW-2.2.0-2.fc36.aarch64.rpm.html

https://fedora.pkgs.org/36/fedora-aarch64/glew-devel-2.2.0-2.fc36.aarch64.rpm.html

I don't use Fedora, but these packages seem promising.

**Comment by Freeplayg** [comment]
Thanks! That seemed to fix that issue, but now I'm missing something else: `Could NOT find SDL (missing: SDL_LIBRARY SDL_INCLUDE_DIR)`

This one is a bit harder for me to find... I've seen people say to do
> `sudo apt-get install libsdl-image1.2-dev`
`sudo apt-get install libsdl-dev.`

but no equivalent for dnf...

**Comment by OposDev** [comment]
Hmmmm.. from looking at the repository, you might be able to fix that by installing either:

https://fedora.pkgs.org/36/fedora-aarch64/SDL_mixer-1.2.12-24.fc36.aarch64.rpm.html
https://fedora.pkgs.org/36/fedora-x86_64/SDL_image-1.2.12-30.fc36.x86_64.rpm.html
https://fedora.pkgs.org/36/fedora-aarch64/SDL_image-1.2.12-30.fc36.aarch64.rpm.html

I'm currently having some issues with the makefile trying to find SDL_mixer.h, but it might be different with this. Hopefully one of these packages work for you! The main dev would know more.

**Comment by Freeplayg** [comment]
They didn't, sadly

**Comment by Freeplayg** [comment]
Oh wait, installing https://fedora.pkgs.org/36/fedora-x86_64/SDL_image-devel-1.2.12-30.fc36.x86_64.rpm.html seemed to fix that! ~~Now for `LZ4 (missing: LZ4_LIBRARY LZ4_INCLUDE_DIR LZ4_GOOD_VERSION)` which I'll try to find...~~ found

**Comment by Freeplayg** [comment]
Now I'm getting a new type of error...
```
CMake Error in CMakeLists.txt:
  Found relative path while evaluating include directories of "wallengine":

    "FFMPEG_AVCODEC_INCLUDE_DIR-NOTFOUND"
```
Which, I'm not sure how to fix this one...

**Comment by OposDev** [comment]
Is FFmpeg installed? 

What about these libraries? 
`sudo apt install libavcodec-dev libavformat-dev libavfilter-dev` (ofc, the fedora variants)

**Comment by Freeplayg** [comment]
Installing those seem to have worked! It said FFmpeg was already installed but I guess I need those, as well. 
But I'm now getting this and I once again... have no idea what to do.

```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
X11_Xxf86vm_LIB (ADVANCED)
    linked by target "wallengine" in directory /home/test/linux-wallpaperengine
```
Thank you for willing to help, btw

**Comment by OposDev** [comment]
I was reading through older issue tickets and came across this:
https://github.com/Almamu/linux-wallpaperengine/issues/72#issuecomment-1104701749
![image](https://user-images.githubusercontent.com/58175621/177168539-066d612d-c1db-40d4-923e-060df77bdef9.png)

Perhaps you have the same issue? From your error, it looks like you're also missing the xxf86vm library package. 

You're welcome :D

**Comment by Freeplayg** [comment]
Installing that worked! But now... I think I'm actually having the same issue you have in #113 ... D:
![image](https://user-images.githubusercontent.com/40742947/177197618-1b8be908-c424-42da-9375-dbfead419619.png)

**Comment by mihawk90** [comment]
Same as in the linked issue, go through the list of requirements in the readme and install the respective development packages.

**Comment by Almamu** [comment]
@Freeplayg please check the requirements specified in the readme and ensure that you have the right (dev/lib) packages for those installed.
